### PR TITLE
cgroup: clean up isIgnorableError for skippable EROFS

### PR DESCRIPTION
**Pull request by cyphar**
Include a rootless argument for isIgnorableError to avoid people
accidentally using isIgnorableError when they shouldn't (we don't ignore
any errors when running as root as that really isn't safe).

Signed-off-by: Aleksa Sarai <asarai@suse.de>

### CODE REVIEW

**Comment by hqhq** [comment]
LGTM

Thanks~

File: libcontainer/cgroups/fs/apply_raw.go [hunk 1]
```diff
@@ -106,11 +106,17 @@ type cgroupData struct {
 // sense of the word). This includes EROFS (which for an unprivileged user is
 // basically a permission error) and EACCES (for similar reasons) as well as
 // the normal EPERM.
-func isIgnorableError(err error) bool {
+func isIgnorableError(rootless bool, err error) bool {
+	// We do not ignore errors if we are root.
+	if !rootless {
```
Comment (rhatdan) [inline libcontainer/cgroups/fs/apply_raw.go]: Should we be smarter and see if you have SYS_ADMIN Caps?

Comment (cyphar) [reply libcontainer/cgroups/fs/apply_raw.go]: That's something that can be handled by having a more clever `isRootless` function earlier on. Not to mention that some of the permission errors would be caused by not having `CAP_DAC_OVERRIDE` -- so `CAP_SYS_ADMIN` checks wouldn't be sufficient.

**Comment by dqminh** [comment]
LGTM

**Pull request merged**